clap_complete/src/dynamic/shells/zsh.rs

```diff
@@ -6,6 +6,45 @@ impl crate::dynamic::Completer for Zsh {
     fn file_name(&self, name: &str) -> String {
         format!("{name}.zsh")
     }
+
+    /// **Generate a completions script into file or stdout for a specified shell at runtime**.
```

Comment:
Why are we documenting a trait method on an impl?  Seems more appropriate to document the trait.

Reply:
Thanks for your review! I did it because different shell scripts have different sourcing methods, so I documented a trait method `write_registration` on an impl, which indeed seems unsuitable. Could you advise where it would be more acceptable to place the shell script sourcing methods? Should I put it at the beginning of zsh.rs, or should I consolidate all the shell script sourcing methods together on the `Completer` trait?

Reply:
This is coupled to https://docs.rs/clap_complete/latest/clap_complete/dynamic/shells/struct.CompleteArgs.html so I would put it on that.

However, there is nuance to this that we need to work through (and could affect the overall design), so let's split this into its own Issue for us to be discussing that nuance and how we want to handle it 

That nuance is this note in #3166

>  Ideally, we have the registration script sourced on-demand in the docs so that it is "auto-updating" with the users tool